kevinChuang/lib/parser.js

```diff
@@ -0,0 +1,21 @@
+/*jshint esversion:6*/
+/*eslint-env es6*/
+
+exports = module.exports = (req,res,next) => {
+  var payload = exports.payload = '';
+  req.on('data',(data)=> {
+    payload += data.toString();
+  });
+
+  req.on('end',()=> {
+    try {
+      var parsedLoad = JSON.parse(payload);
+      req.body = parsedLoad;
+    } catch (err) {
+      err.message = 'invalid Jayson';
+      err.statusCode = 422;
+      next(err);
```

Comment:
Don't forget to return this.  Otherwise you may end up with an unhandled error event.  
Like this --> return next(err)